pkg/manifests/testdata/artifacts/blobs-only/blobs/sha256/579b4ca613dbc3017ec77e006dc9a275e074591d1c19b635e49aaf2dd07cfe3d

```diff
@@ -0,0 +1,17 @@
+{
+  "schemaVersion": 2,
+  "mediaType": "application/vnd.oci.image.manifest.v1+json",
+  "artifactType": "application/vnd.example+type",
```

Comment:
Which artifact type is supported? If I run a simple sample like this:

```go
	storeOpts, err := cstorage.DefaultStoreOptions()
	if err != nil {
		panic(err)
	}
	store, err := cstorage.GetStore(storeOpts)
	if err != nil {
		panic(err)
	}
	runtime, err := libimage.RuntimeFromStore(store, &libimage.RuntimeOptions{})
	if err != nil {
		panic(err)
	}
	_, err = runtime.Pull(context.Background(), "quay.io/saschagrunert/seccomp:v1", commonconfig.PullPolicyAlways, &libimage.PullOptions{})
	if err != nil {
		panic(err)
	}
```

Then I get the panic:

```
panic: parsing image configuration: unsupported image-specific operation on artifact with type "application/vnd.unknown.artifact.v1"
```

The same for `quay.io/saschagrunert/seccomp:v2`:

```
panic: parsing image configuration: unsupported image-specific operation on artifact with type "application/vnd.example+type"
```

Using commit 1ad13e9459e3 and github.com/containers/image/v5 v5.29.3-0.20240131175401-a63f4a542670

Reply:
This PR doesn't do anything for pulling artifacts into local storage.  Its main concern is adding the ability to craft artifact manifests for local files and add them to manifest lists, and broadening support for preserving information like `artifactType` fields.